src/analyzer.ts

```diff
@@ -97,6 +101,10 @@ export class Analyzer {
     return {ok: true, value: rootConfig};
   }
 
+  async #readPackageJson(packageDir: string): Promise<Result<PackageJson>> {
```

Comment:
Maybe just inline this, since it's only used in one place and is trivial and already readable.